Sources/NIO/SocketAddresses.swift

```diff
@@ -264,27 +264,35 @@ public enum SocketAddress: CustomStringConvertible {
     /// - returns: the `SocketAddress` corresponding to this string and port combination.
     /// - throws: may throw `SocketAddressError.failedToParseIPString` if the IP address cannot be parsed.
     public init(ipAddress: String, port: Int) throws {
-        var ipv4Addr = in_addr()
-        var ipv6Addr = in6_addr()
-
         self = try ipAddress.withCString {
-            if inet_pton(NIOBSDSocket.AddressFamily.inet.rawValue, $0, &ipv4Addr) == 1 {
+            do {
+                var ipv4Addr = in_addr()
+                try NIOBSDSocket.inet_pton(af: .inet, src: $0, dst: &ipv4Addr)
+
                 var addr = sockaddr_in()
                 addr.sin_family = sa_family_t(NIOBSDSocket.AddressFamily.inet.rawValue)
                 addr.sin_port = in_port_t(port).bigEndian
                 addr.sin_addr = ipv4Addr
+
                 return .v4(.init(address: addr, host: ""))
-            } else if inet_pton(NIOBSDSocket.AddressFamily.inet6.rawValue, $0, &ipv6Addr) == 1 {
+            } catch {
```

Comment:
empty `catch` block

Reply:
The empty `catch` is much nicer to follow - it prevents the embedded, deeply nested approach.  Is there a better way to write this?  The `guard ...` approach is just as nested.

Reply:
If you're going to do that you have to put a comment in it pointing out that it's intentional.

Reply:
Sure, happy to add a comment, that request wasn't clear.  If there is a better way to write this, I'm also okay with that, I just don't know of one.